servers/republik/modules/crowdfundings/lib/scheduler/changeover.js

```diff
@@ -0,0 +1,146 @@
+      INNER JOIN "membershipPeriods" mp
+        ON m.id = mp."membershipId"
+      INNER JOIN "membershipTypes" mt
+        ON m."membershipTypeId" = mt.id
+        AND mt.name != 'MONTHLY_ABO' -- Exclude subscription via Stripe
+      WHERE
+        m.active = true
+      GROUP BY 1
+    ) AS users
+    WHERE
+      -- Potential ending memberships to change over to dormant membership
+      ( "endDate" < :endDate )
+    GROUP BY 1
+  `, { endDate })
+
+  debug({
+    now: endDate.toDate(),
+    users: users.length
+  })
+
+  const stats = { changeover: 0 }
+
+  await Promise.each(
+    users,
+    async user => {
+      // Find all memberships a user owns
+      let userMemberships = await pgdb.public.memberships.find(
+        { userId: user.id }
+      )
+
+      userMemberships = await resolveMemberships({
+        memberships: userMemberships,
+        pgdb
+      })
+
+      if (userMemberships.filter(m => m.active).length > 1) {
+        console.error(
+          'changeover failed: multiple active memberships found',
+          { user }
+        )
+        return
+      }
+
+      const activeMembership = userMemberships.find(m => m.active)
+      const dormantMemberships = findDormantMemberships({
+        memberships: userMemberships,
+        user: { id: user.id }
+      })
+
+      // End here if there is not dormant membership to be found
+      if (dormantMemberships.length < 1) {
+        return
+      }
+
+      // Elect a dormant membership to activate. Rule is to elect dorman
+      // membership with lowest sequenceNumber
+      const electedDormantMembership = dormantMemberships.reduce(
+        (acc, curr) =>
+          !acc || curr.sequenceNumber < acc.sequenceNumber
+            ? curr
+            : acc
+      )
+
+      const transaction = await pgdb.transactionBegin()
+      debug({
+        activeMembership: activeMembership.id,
+        electedDormantMembership: electedDormantMembership.id
+      })
+
+      try {
+        const now = moment()
+
+        await transaction.public.memberships.update(
+          { id: activeMembership.id },
+          { active: false, renew: false, updatedAt: now }
+        )
+
+        await transaction.public.memberships.updateOne(
+          { id: electedDormantMembership.id },
+          {
+            active: true,
+            renew: true,
+            sequenceNumber: activeMembership.sequenceNumber,
+            updatedAt: now
+          }
+        )
+
+        const beginDate = now
+        const endDate = beginDate.clone().add(
+          electedDormantMembership.initialPeriods,
+          electedDormantMembership.initialInterval
+        )
+
+        await transaction.public.membershipPeriods.insert({
+          membershipId: electedDormantMembership.id,
+          beginDate,
+          endDate,
+          kind: 'CHANGEOVER',
+          createdAt: now,
+          updatedAt: now
+        })
+
+        await enforceSubscriptions({ userId: user.id, pgdb })
```

Comment:
put this after `transactionCommit` or pass `transaction as pgdb` to it, otherwise the code there doesn't see the changes just made